### ISSUE TITLE
Building package from source

### ISSUE THREAD

**Issue by rkubosz** [Abjad/abjad#912]
Hej!
I maintain the package with abjad in AUR -- Arch Linux user's repository. The package has to install v2.21 release of abjad.
AUR's policy recommends to build packages from source. I tried to build and install the v2.21 with `sudo python setup.py install 2>install_log_err.txt`, but I get some confusing error log (attached in file).
[install_log_err.txt](https://github.com/Abjad/abjad/files/1671581/install_log_err.txt)

Am I missing some dependencies?

**Comment by josiah-wolf-oberholtzer** [comment]
The technique you're using is trying to compile some template files which are not valid Python (and therefore should not be compiled).

Maybe try `python setup.py sdist` to create a distribution? That doesn't error, and it's the command we use to create the official package for upload to PyPI.

(Off topic: why would you want to install from the Arch repository and not PyPI?)

**Comment by rkubosz** [comment]
Thanks for answer! I changed `python setup.py install --root="$pkgdir/" --optimize=1` to `python setup.py install --root="$pkgdir/" --no-compile`in my PKGBUILD. Now it installs without these errors.

I've decided to distribute Abjad with AUR since Python packages are distributed in Arch repositories (official and in AUR) and Arch developers recommend install Python packages from these repos at first place, then eventually to try PyPI. [[source]](https://wiki.archlinux.org/index.php/Python#Package_management)

IMHO AUR package is a totally fine way for users to install Abjad and for basic use: to look what it is, to fool around with it and to use it occasionally. But still for more intensive and project-oriented use it should be installed in virtalenv via PyPI, like you recommend.

I am closing this issue. Thanks for help!
